test/strict/classComputedPropertyName.js

```diff
@@ -0,0 +1,245 @@
+                        static [a < 6 ? delete Object.prototype : 5]() { }
+                    }
+                },
+                TypeError,
+                "Computed property names inside classes are specified to execute in strict mode, \
+                thus calling delete on an undeletable property of object should throw a TypeError \
+                in strict mode",
+                "Calling delete on 'prototype' is not allowed in strict mode"
+            );
+            assert.throws(
+                function () {
+                    var a = {};
+                    Object.preventExtensions(a);
+                    class C {
+                        [a && delete Object.prototype]() { }
+                    }
+                },
+                TypeError,
+                "Computed property names inside classes are specified to execute in strict mode, \
+                thus calling delete on an undeletable property of object should throw a TypeError \
+                in strict mode",
+                "Calling delete on 'prototype' is not allowed in strict mode"
+            );
+            assert.throws(
+                function () {
+                    var a = {};
+                    Object.preventExtensions(a);
+                    class C {
+                        static [a && delete Object.prototype]() { }
+                    }
+                },
+                TypeError,
+                "Computed property names inside classes are specified to execute in strict mode, \
+                thus calling delete on an undeletable property of object should throw a TypeError \
+                in strict mode",
+                "Calling delete on 'prototype' is not allowed in strict mode"
+            );
+            assert.throws(
+                function () {
+                    var a = {};
+                    Object.defineProperty(a, "x", { value: 5, configurable: false });
+                    class C {
+                        [delete a["x"]]() { }
+                    }
+                },
+                TypeError,
+                "Computed property names inside classes are specified to execute in strict mode, \
+                thus calling delete on an undeletable property of object should throw a TypeError \
+                in strict mode",
+                "Calling delete on 'x' is not allowed in strict mode"
+            );
+            assert.throws(
+                function () {
+                    var a = {};
+                    Object.defineProperty(a, "x", { value: 5, configurable: false });
+                    class C {
+                        static [delete a["x"]]() { }
+                    }
+                },
+                TypeError,
+                "Computed property names inside classes are specified to execute in strict mode, \
+                thus calling delete on an undeletable property of object should throw a TypeError \
+                in strict mode",

```

Comment:
nit: you can use strings with `message` to do multi-line strings, unless you specifically wanted this to print out as a single line with lots of spaces in the middle:

```
> "hello \
...       world"
'hello       world'
```

(Probably doesn't matter because this is only printed in case of incorrect behavior.)

Reply:
84292eb